### PR TITLE
EDSC-2216: Implements order status jobs

### PR DESCRIPTION
**Pull request by abbottry**

### CODE REVIEW

File: serverless/src/submitCatalogRestOrder/handler.js [hunk 1]
```diff
@@ -115,17 +120,20 @@ const submitCatalogRestOrder = async (event, context) => {
...
     Object.keys(orderPayload)
-      .forEach(key => (orderPayload[key].length === 0) && delete orderPayload[key])
+      .forEach(key => (orderPayload[key] == null
+        || orderPayload[key].length === 0) && delete orderPayload[key])
 
     try {
       const orderResponse = await request.post({
         uri: url,
         form: orderPayload,
+
```
Comment (macrouch) [inline serverless/src/submitCatalogRestOrder/handler.js]: weird spot for a empty line

File: serverless/src/util/echoForms/getEmail.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { getFieldElementValue } from './getFieldElementValue'
+
+/**
+ * Get the email address from the provided XML Document
+ * @param {String} xmlDocument ECHO Form xml as a string
+ */
+export const getEmail = (xmlDocument) => {
```
Comment (macrouch) [inline serverless/src/util/echoForms/getEmail.js]: seems like an easy method to write a test for

File: static/src/js/util/orders.js [hunk 3]
```diff
@@ -3,9 +3,9 @@ import { cmrEnv } from '../../../../sharedUtils/cmrEnv'
 
 
 /**
- * Prepare parameters used in submitOrders() based on current Redux State
+ * Prepare parameters used in submitRetrieval() based on current Redux State
  * @param {object} state Current Redux State
- * @returns Parameters used in submitOrders()
+ * @returns Parameters used in submitRetrieval()
  */
 export const prepareOrderParams = (state) => {
```
Comment (macrouch) [inline static/src/js/util/orders.js]: should this be `prepareRetrievalParams`? and maybe rename the file?

File: serverless/src/submitLegacyServicesOrder/handler.js [hunk 4]
```diff
@@ -125,6 +132,9 @@ const submitLegacyServicesOrder = async (event, context) => {
       })
 
       await dbConnection('retrieval_orders').update({ order_number: orderId }).where({ id })
+
+      // start the order status check workflow
```
Comment (trevorlang) [inline serverless/src/submitLegacyServicesOrder/handler.js]: Capitalization :)

File: static/src/js/components/OrderProgressList/OrderProgressItem.js [hunk 5]
```diff
@@ -39,7 +43,7 @@ export const OrderProgressItem = ({
         </h5>
         <div className="order-progress-item__info">
           <span className="order-progress-item__processed">
-            {`${totalProcessed} of ${totalNumber} granules processed (${percentProcessed}%)`}
+            { `${totalProcessed} of ${totalNumber} granule(s) processed (${totalPercentProcessed}%)`}
```
Comment (trevorlang) [inline static/src/js/components/OrderProgressList/OrderProgressItem.js]: Extra space between curly and `

File: static/src/js/components/OrderProgressList/OrderProgressList.js [hunk 6]
```diff
@@ -6,14 +6,18 @@ import OrderProgressItem from './OrderProgressItem'
...
+  orders,
+  totalNumber,
+  totalProcessed
 }) => (
   <ul className="order-progress-list">
     {
       orders.map(order => (
         <OrderProgressItem
-          key={order.order_id}
+          key={order.order_number}
```
Comment (trevorlang) [inline static/src/js/components/OrderProgressList/OrderProgressList.js]: Destructure maybe?

File: static/src/js/components/OrderStatus/OrderStatusItem.js [hunk 7]
```diff
@@ -1,51 +1,93 @@
...
-      'order-status-item--in-progress': getStateFromOrderStatus(orderStatus) === 'in progress',
-      'order-status-item--errored': getStateFromOrderStatus(orderStatus) === 'errored'
+        // Start refreshing the retrieval collection every 20 seconds
+        setInterval(() => {
+          try {
+            onFetchRetrievalCollection(id)
+          } catch (e) {
+            console.log(e)
+          }
+        }, 60000)
```
Comment (trevorlang) [inline static/src/js/components/OrderStatus/OrderStatusItem.js]: Would it make sense to abstract these delay times out into the config somewhere?

File: static/src/js/constants/actionTypes.js [hunk 8]
```diff
@@ -79,6 +79,8 @@ export const ADD_ACCESS_METHODS = 'ADD_ACCESS_METHODS'
 export const SELECT_ACCESS_METHOD = 'SELECT_ACCESS_METHOD'
 export const UPDATE_ACCESS_METHOD = 'UPDATE_ACCESS_METHOD'
 
-export const UPDATE_ORDER = 'UPDATE_ORDER'
+export const UPDATE_RETRIEVAL = 'UPDATE_RETRIEVAL'
+
```
Comment (trevorlang) [inline static/src/js/constants/actionTypes.js]: Could probably remove this space

**Pull request merged**